docs/specification/sep/sep-client-id-metadata.md

```diff
@@ -0,0 +1,270 @@
+- Servers SHOULD cache metadata respecting HTTP headers (max 24 hours recommended)
+- Servers MUST validate redirect URIs match those in metadata document
+
+### Discovery
+- Servers advertise support via OAuth metadata: `client_id_metadata_document_supported: true`
+- Clients detect support and can fallback to DCR or pre-registration if unavailable
+
+
+Example metadata document:
+```json
+{
+  "client_id": "https://app.example.com/oauth/client-metadata.json",
+  "client_name": "Example MCP Client",
+  "client_uri": "https://app.example.com",
+  "logo_uri": "https://app.example.com/logo.png",
+  "redirect_uris": [
+    "http://127.0.0.1:3000/callback",
+    "http://localhost:3000/callback"
+  ],
+  "grant_types": ["authorization_code"],
+  "response_types": ["code"],
+  "token_endpoint_auth_method": "none"
+}
+```
+
+
+### Integration with Existing MCP Auth
+
+This proposal adds Client ID Metadata Documents as a third registration option alongside pre-registration and DCR. Servers MAY support any combination of these approaches:
+
+- Pre-registration remains unchanged
+- DCR remains unchanged
+- Client ID Metadata Documents are detected by URL-formatted client_ids, and server support is advertised in OAuth metadata.
+
+## Rationale
+
+### Why This Solves the "No Pre-existing Relationship" Problem
+
+Unlike pre-registration which requires coordination, or DCR which requires servers to manage a registration database, Client ID Metadata Documents provide:
+
+1. **Verifiable Identity**: The HTTPS URL serves as both identifier and trust anchor
+2. **No Coordination Needed**: Clients publish metadata, servers consume it
+3. **Flexible Trust Policies**: Servers decide their own trust criteria without requiring client changes
+4. **Stable Identifiers**: Unlike DCR's ephemeral IDs, URLs are stable and auditable
+
+### Redirect URI Attestation
+
+A key benefit of Client ID Metadata Documents is attestation of redirect URIs:
+
+1. **The metadata document cryptographically binds redirect URIs to the client identity** via HTTPS
+2. **Servers can trust that redirect URIs in the metadata are controlled by the client** - not attacker-supplied
+3. **This prevents redirect URI manipulation attacks** common with self-asserted registration
+
+
+### Risks of this approach
+
+#### Risk: Localhost URL Impersonation
+
+A limitation of Client ID Metadata Documents is that they cannot prevent localhost URL impersonation by itself. An attacker can claim to be any client by:
+1. Providing the legitimate client's metadata URL as their client_id
+2. Binding to the same localhost port the legitimate client uses
+3. Intercepting the authorization code when the user approves
+
+This attack is concerning because the server sees the correct metadata
+document and the user sees the correct client name, making detection
+difficult. While platform-specific attestations (iOS DeviceCheck, Android 
+Play Integrity) could address this, they're not universally available. An 
```

Comment:
Let's expand on this and describe how an app developer can run a backend service that validates the DeviceCheck/Play Integrity signatures and returns a JWT the app can use with the `private_key_jwt` client authentication method. This would actually be a pretty reasonable level of assurance of the app's identity. This doesn't need to be specified normatively in this proposal, since the actual underlying OAuth mechanism used is only the `private_key_jwt` auth method.

Reply:
Perhaps the normative text can be left in the IETF draft, and we can just normatively reference that instead and outline the concept of using "private_key_jwt" here. Alternatively we can add a section to talk about client authentication as a core part of the spec, and not just as a mitigation in this section.